internal/services/configuration/auth/service.go

```diff
@@ -0,0 +1,220 @@
+	}
+
+	service := &Service{
+		logger:   appCtx.Logger,
+		Provider: config.LoadOCIConfig(),
+	}
+	logger.LogWithLevel(service.logger, 3, "Created new authentication service")
+	return service
+}
+
+// PromptForProfile prompts the user to select an OCI profile.
+func (s *Service) PromptForProfile() (string, error) {
+	logger.LogWithLevel(s.logger, 3, "Prompting user for OCI profile selection")
+
+	fmt.Println("Do you want to use the DEFAULT profile or enter a custom profile name?")
+	fmt.Println("1: Use DEFAULT profile")
+	fmt.Println("2: Enter custom profile name")
+	fmt.Print("Enter your choice (1 or 2): ")
+	reader := bufio.NewReader(os.Stdin)
+	choice, err := reader.ReadString('\n')
+	if err != nil {
+		return "", errors.Wrap(err, "reading profile choice input")
+	}
+
+	choice = strings.TrimSpace(choice)
+
+	logger.LogWithLevel(s.logger, 3, "User selected profile choice", "choice", choice)
+
+	profile := "DEFAULT"
+	if choice == "2" {
+		fmt.Print("Enter profile name: ")
+		customProfile, err := reader.ReadString('\n')
+
+		if err != nil {
+			return "", errors.Wrap(err, "reading custom profile input")
+		}
+
+		profile = strings.TrimSpace(customProfile)
+
+		logger.LogWithLevel(s.logger, 1, "Using custom profile", "profile", profile)
+		fmt.Printf("Using profile: %s\n", profile)
+	} else {
+		logger.LogWithLevel(s.logger, 1, "Using DEFAULT profile")
+		fmt.Println("Using DEFAULT profile")
+	}
+
+	return profile, nil
+}
+
+// GetOCIRegions returns a list of all available OCI regions.
+func (s *Service) GetOCIRegions() []RegionInfo {
+	logger.LogWithLevel(s.logger, 3, "Getting list of OCI regions")
+
+	regions := []string{
+		"af-johannesburg-1", "ap-batam-1", "ap-chiyoda-1", "ap-chuncheon-1",
+		"ap-chuncheon-2", "ap-dcc-canberra-1", "ap-dcc-gazipur-1", "ap-hyderabad-1",
+		"ap-ibaraki-1", "ap-melbourne-1", "ap-mumbai-1", "ap-osaka-1", "ap-seoul-1",
+		"ap-seoul-2", "ap-singapore-1", "ap-singapore-2", "ap-suwon-1", "ap-sydney-1",
+		"ap-tokyo-1", "ca-montreal-1", "ca-toronto-1", "eu-amsterdam-1", "eu-crissier-1",
+		"eu-dcc-dublin-1", "eu-dcc-dublin-2", "eu-dcc-milan-1", "eu-dcc-milan-2",
+		"eu-dcc-rating-1", "eu-dcc-rating-2", "eu-dcc-zurich-1", "eu-frankfurt-1",
+		"eu-frankfurt-2", "eu-jovanovac-1", "eu-madrid-1", "eu-madrid-2",
+		"eu-marseille-1", "eu-milan-1", "eu-paris-1", "eu-stockholm-1",
+		"eu-zurich-1", "il-jerusalem-1", "me-abudhabi-1", "me-abudhabi-2",
+		"me-abudhabi-3", "me-abudhabi-4", "me-alain-1", "me-dcc-doha-1",
+		"me-dcc-muscat-1", "me-dubai-1", "me-jeddah-1", "me-riyadh-1",
+		"mx-monterrey-1", "mx-queretaro-1", "sa-bogota-1", "sa-santiago-1",
+		"sa-saopaulo-1", "sa-valparaiso-1", "sa-vinhedo-1", "uk-cardiff-1",
+		"uk-gov-cardiff-1", "uk-gov-london-1", "uk-london-1", "us-abilene-1",
+		"us-ashburn-1", "us-chicago-1", "us-dallas-1", "us-gov-ashburn-1",
+		"us-gov-chicago-1", "us-gov-phoenix-1", "us-langley-1", "us-luke-1",
+		"us-phoenix-1", "us-saltlake-2", "us-sanjose-1", "us-somerset-1",
+		"us-thames-1",
```

Comment:
Should we be hard-coding these as they change over time? Can we not pull them down with an API call?